spec/forms/course_detail_form_spec.rb

```diff
@@ -320,6 +320,15 @@
           .and change { trainee.course_end_date }
           .from(nil).to(Date.parse(valid_end_date.to_s))
       end
+
+      context "when a trainee has a course code" do
+        let(:trainee) { create(:trainee, course_code: Faker::Alphanumeric.alphanumeric(number: 4).upcase) }
+        it "doesnt wipe course code" do
```

Comment:
can we space this and move the `subject.save` and `trainee.reload` into a before block to organise